src/cplus_plugin/settings.py

```diff
@@ -0,0 +1,243 @@
+            )
+
+    def save_settings(self) -> None:
+        """Saves the settings.
+        Also does error checking for settings (e.g if the custom logo exists).
+        Will present the user with an error message if an issue is found.
+        """
+
+        # Analysis configuration
+
+        # Report settings
+        organization = self.txt_organization.text()
+        settings_manager.set_value(Settings.REPORT_ORGANIZATION, organization)
+
+        email = self.txt_email.text()
+        settings_manager.set_value(Settings.REPORT_CONTACT_EMAIL, email)
+
+        website = self.txt_website.text()
+        settings_manager.set_value(Settings.REPORT_WEBSITE, website)
+
+        custom_logo = self.cb_custom_logo.checkState()
+        settings_manager.set_value(Settings.REPORT_CUSTOM_LOGO, custom_logo)
+
+        # Checks if the logo file exists if custom logo is enabled
+        if custom_logo > 0:
+            custom_logo_path = self.logo_file.filePath()
+            settings_manager.set_value(Settings.REPORT_LOGO_DIR, custom_logo_path)
+
+            if not os.path.exists(custom_logo_path):
+                # File not found, disable custom logo
+                settings_manager.set_value(Settings.REPORT_CUSTOM_LOGO, 0)
+
+                iface.messageBar().pushWarning(
+                    "CPLUS - Custom logo not found, disabled: ", custom_logo_path
+                )
+
+        footer = self.txt_footer.toPlainText()
+        settings_manager.set_value(Settings.REPORT_FOOTER, footer)
+
+        disclaimer = self.txt_disclaimer.toPlainText()
+        settings_manager.set_value(Settings.REPORT_DISLAIMER, disclaimer)
+
+        report_license = self.txt_license.text()
+        settings_manager.set_value(Settings.REPORT_LICENSE, report_license)
+
+        # Advanced settings
+        base_dir_path = self.folder_data.filePath()
+        settings_manager.set_value(Settings.BASE_DIR, base_dir_path)
+
+        # Checks if the provided base directory exists
+        if not os.path.exists(base_dir_path):
+            iface.messageBar().pushCritical(
+                "CPLUS - Base directory not found: ", base_dir_path
+            )
+
+    def load_settings(self) -> None:
+        """Loads the settings and displays it in the options UI"""
+
+        # Analysis configuration settings
+
+        # Report settings
+        organization = settings_manager.get_value(
+            Settings.REPORT_ORGANIZATION, default=""
+        )
+        self.txt_organization.setText(organization)
+
+        email = settings_manager.get_value(Settings.REPORT_CONTACT_EMAIL, default="")
+        self.txt_email.setText(email)
+
+        website = settings_manager.get_value(Settings.REPORT_WEBSITE, default="")
+        self.txt_website.setText(website)
+
+        custom_logo = int(
+            settings_manager.get_value(
+                Settings.REPORT_CUSTOM_LOGO,
+                default=0,  # False is 0 for the checkbox state
```

Comment:
`False` default value with https://doc.qt.io/qt-5/qabstractbutton.html#checked-prop can be used for a direct approach instead of the current used `0`

Reply:
Done